### PR TITLE
Introduce a new WebKit checker for a unchecked call arguments (#113708)

### PR DESCRIPTION
**Pull request by rniwa**
This PR introduces alpha.webkit.UncheckedCallArgsChecker which detects a function argument which is a raw reference or a raw pointer to a CheckedPtr capable object.

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h [hunk 1]
```diff
@@ -55,10 +55,18 @@ bool isCheckedPtr(const clang::CXXRecordDecl *Class);
 /// not, std::nullopt if inconclusive.
 std::optional<bool> isUncounted(const clang::QualType T);
 
+/// \returns true if \p Class is CheckedPtr capable AND not checked, false if
+/// not, std::nullopt if inconclusive.
+std::optional<bool> isUnchecked(const clang::QualType T);
```
Comment (t-rasmud) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h:60]: I can't seem to find a reference to this overload of `isUnchecked`. Do you see a use of it in future?

File: clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefCallArgsChecker.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//=======- UncountedCallArgsChecker.cpp --------------------------*- C++ -*-==//
+//=======- RawPtrRefCallArgsChecker.cpp --------------------------*- C++ -*-==//
```
Comment (t-rasmud) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefCallArgsChecker.cpp:1]: Would it make sense to have documentation for `RawPtrCallArgsChecker` in `checkers.rst`? Do you anticipate more generic checkers calling this checker besides `UncountedCallArgsChecker` and `UncheckedCallArgsChecker`?

**Review by t-rasmud — APPROVED** [review]
I had a couple clarifying questions, otherwise LGTM!

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h (hunk 1 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h:60]: `UncheckedCallArgsChecker::isUnsafeType` calls it.

File: clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefCallArgsChecker.cpp (hunk 2 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/RawPtrRefCallArgsChecker.cpp:1]: I don't think so. I could imagine `UncountedCallArgsChecker` and `UncheckedCallArgsChecker` getting merged in the future but I don't think we'd add a new checker using `RawPtrCallArgsChecker`.

**Comment by rniwa** [comment]
Thanks for the review!

**Pull request merged**